File: src/main/java/com/github/shatteredsuite/scrolls/data/scroll/UnboundBindingType.java
package com.github.shatteredsuite.scrolls.data.scroll;

import com.github.shatteredsuite.core.include.nbt.NBTCompound;
import java.util.LinkedHashMap;
import java.util.Map;
import org.jetbrains.annotations.NotNull;

public class UnboundBindingType extends BindingType {
    public UnboundBindingType(String key) {
        super("location");
    }

    @Override
    public BindingData deserialize(Map<String, Object> map) {
        return null;
    }

    @Override
    public BindingData fromNBT(NBTCompound compound) {
        return new UnboundBindingData();
    }
}

File: src/main/kotlin/com/github/shatteredsuite/scrolls/data/scroll/binding/UnboundBindingType.kt
package com.github.shatteredsuite.scrolls.data.scroll.binding

import com.github.shatteredsuite.core.include.nbt.NBTCompound
import org.bukkit.command.CommandSender

class UnboundBindingType : BindingType("unbound") {
    override fun deserialize(map: Map<String?, Any?>): BindingData {
        return UnboundBindingData()
    }

    override fun createFromCommandArgs(args: Array<out String>, sender: CommandSender): BindingData {
        return UnboundBindingData()
    }

    override fun tabCompleteCommandArgs(args: Array<out String>, sender: CommandSender): MutableList<String> {
        return mutableListOf()
    }

    override fun fromNBT(compound: NBTCompound): BindingData {
        return UnboundBindingData()
    }
}